src/Equations/EffectiveCriticalField.cpp

```diff
@@ -57,6 +80,52 @@ bool EffectiveCriticalField::GridRebuilt(){
         ECRIT_POPTIMUM_PREV[ir] = 10;
     }
 
+        // placeholder quantities that will be overwritten by the GSL functions. Initialize here
+    // so we can use previous values
+    std::function<real_t(real_t,real_t,real_t)> Func = [](real_t,real_t,real_t){return 0;};
+    gsl_parameters.Func = Func; 
+    gsl_parameters.Eterm = 0;
+    gsl_parameters.p = 0;
+    gsl_parameters.p_ex_lo = 0;
+    gsl_parameters.p_ex_up = 0; 
+
+
+    if ((Eceff_mode == OptionConstants::COLLQTY_ECEFF_MODE_SIMPLE) || (Eceff_mode == OptionConstants::COLLQTY_ECEFF_MODE_FULL)){
+        
+        this->EOverUnityContrib = new real_t*[nr];
+        this->SynchOverUnityContrib = new real_t*[nr];
+
+        this->gsl_parameters.EContribSpline = new gsl_spline*[nr]; 
+        this->gsl_parameters.SynchContribSpline = new gsl_spline*[nr];
+
+        this->gsl_parameters.EContribAcc = gsl_interp_accel_alloc(); // the accelerators cache values from the splines
+        this->gsl_parameters.SynchContribAcc = gsl_interp_accel_alloc();
```

Comment:
I don't quite remember (and I've previously stated otherwise), but I think these accelerator objects should only be assigned to one spline each.. so for example the EContribAcc should be a pointer to an array of nr accelerators

Reply:
actually, I first thought you were wrong and implemented arrays of accelerators, but then I realized they could be reset so I thought that was what you meant all the time;) Isn't resetting in every radial point better? If I need to re-calculate Eceff I need to reset them anyway, so no need to save for later iterations.

Reply:
I thought you only needed to reset if the data behind the splines has changed?

Reply:
The accelerators function as "memory" for looking up points on the x grid (in this case r?). When you reset the accelerator you wipe the memory which has been built during the past evaluations of the spline. Accelerators are useful because, aside from just multiplying/adding the spline coefficients, the other expensive part of the evaluation is to figure which set of coefficients (corresponding to one/several x points) to use. Since we can generally make very few assumptions about the x vector (I think GSL only assumes it is monotonically increasing/decreasing), these search algorithms tend to be very general and slow, which is why accelerators can make a big difference when evaluating splines.

Reply:
@Embreus yes, the data changes because I the same accelerator for all radii. So instead of a new accelerator I just reset it between different radial points. 
@hoppe93: we interpolate in A here

Reply:
I think it would be a good idea then to have arrays of accelerators, to avoid having to reset them.